app/src/main/java/com/github/libretube/util/NewPipeDownloaderImpl.kt

```diff
@@ -11,50 +9,48 @@ import org.schabi.newpipe.extractor.downloader.Response
 import org.schabi.newpipe.extractor.exceptions.ReCaptchaException
 
 class NewPipeDownloaderImpl : Downloader() {
-    private val client: OkHttpClient = OkHttpClient.Builder()
-        .readTimeout(READ_TIMEOUT_SECONDS, TimeUnit.SECONDS)
+    private val client = OkHttpClient.Builder()
         .build()
 
     @Throws(IOException::class, ReCaptchaException::class)
     override fun execute(request: Request): Response {
+        val httpMethod = request.httpMethod()
         val url = request.url()
-
-        val requestBody = request.dataToSend()?.let {
-            it.toRequestBody(APPLICATION_JSON, 0, it.size)
-        }
+        val headers = request.headers()
+        val dataToSend = request.dataToSend()
 
         val requestBuilder = okhttp3.Request.Builder()
-            .method(request.httpMethod(), requestBody)
+            .method(httpMethod, dataToSend?.toRequestBody())
             .url(url)
-            .addHeader(USER_AGENT_HEADER_NAME, USER_AGENT)
+            .addHeader("User-Agent", USER_AGENT)
 
-        for ((headerName, headerValueList) in request.headers()) {
-            requestBuilder.removeHeader(headerName)
-            for (headerValue in headerValueList) {
+        headers.forEach { (headerName, headerValueList) ->
+            headerValueList.forEach { headerValue ->
```

Comment:
Using normal for loops is completely valid as well and easier to read in my opinion.